share/lib/python/neuron/__init__.py

```diff
@@ -148,6 +148,57 @@
 
 config._parse_arguments(h)
 
+
+def _check_for_intel_openmp():
+    """Check if Intel's OpenMP runtime has already been loaded.
+
+    This does not interact well with the NVIDIA OpenMP runtime in CoreNEURON GPU
+    builds. See
+    https://forums.developer.nvidia.com/t/nvc-openacc-runtime-segfaults-if-intel-mkl-numpy-is-already-loaded/212739
+    for more information.
+    """
+    import ctypes
+    from neuron.config import arguments
+
+    # These checks are only relevant for shared library builds with CoreNEURON GPU support enabled.
+    if (
+        not arguments["NRN_ENABLE_CORENEURON"]
+        or not arguments["CORENRN_ENABLE_GPU"]
+        or not arguments["CORENRN_ENABLE_SHARED"]
+    ):
+        return
+
+    current_exe = ctypes.CDLL(None)
+    try:
+        # Picked quasi-randomly from `nm libiomp5.so`
+        current_exe["_You_must_link_with_Intel_OpenMP_library"]
```

Comment:
👁️ Seems like they haven't changed this often - https://github.com/intel/llvm/blame/fab6b1af6e11efbefa40b849925179a1b7c82650/openmp/runtime/src/kmp_import.cpp